src/particles/elements/DipEdge.H

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2021 Chad Mitchell, Axel Huebl
+ *
+ * This file is part of ImpactX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_DIPEDGE_H
+#define IMPACTX_DIPEDGE_H
+
+#include "particles/ImpactXParticleContainer.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+
+#include <cmath>
+
+
+namespace impactx
+{
+    struct DipEdge
+    {
+        using PType = ImpactXParticleContainer::ParticleType;
+
+        /** Edge focusing associated with bend entry or exit
```

Comment:
Awesome! Don't forget to add in the user-facing docs please:
- https://impactx.readthedocs.io/en/latest/usage/parameters.html#lattice-elements
- aka: `docs/source/usage/parameters.rst`

Reply:
Upcoming in next commit.